### ISSUE TITLE
WASM backend fails to initialize on node 18

### ISSUE THREAD

**Issue by jkrems** [tensorflow/tfjs#6857]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow.js):
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow.js installed from (npm or script link):
- TensorFlow.js version (use command below):
- Browser version:
- Tensorflow.js Converter Version:

**Describe the current behavior**

When running in node 18, the WASM backend gets confused by the existence of `globalThis.fetch` and assumes it works as a drop-in for the fetch logic used in older versions of node. But the standard fetch in node won't load file system resources, breaking existing code running in newer versions of node.

```js
const tf = require('@tensorflow/tfjs-core');
const {setWasmPath} = require('@tensorflow/tfjs-backend-wasm/dist/tf-backend-wasm.min.js');

const wasmPath = require.resolve('@tensorflow/tfjs-backend-wasm/wasm-out/tfjs-backend-wasm.wasm');
setWasmPath(wasmPath);
await tf.setBackend('wasm');
// Initialization of backend wasm failed
// TypeError: Failed to parse URL from /private/tmp/tfjs-example/node_modules/@tensorflow/tfjs-backend-wasm/wasm-out/tfjs-backend-wasm.wasm
```

**Describe the expected behavior**

WASM payload is loaded from file system, even in versions of node with a global `fetch` for loading HTTP resources.

**Standalone code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate
the problem. If possible, please share a link to Colab/CodePen/any notebook.

**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.

**Comment by mattsoulanille** [comment]
Thanks for the bug report! I've confirmed your test code works on Node v17.9.1 but not on v18.9.1.

**Comment by mattsoulanille** [comment]
I just tried testing against the main branch (I was testing against 3.20.0 above), and this appears to have been fixed. You can test this yourself by putting your code snippet in the file `link-package/index.js` and running the following (on the master branch):

```bash
yarn # In the root of the repository
cd link-package
yarn build-deps-for tfjs-backend-wasm # Build the npm artifact for the wasm backend and link it in the link package
node index.js
```

Please let me know if this doesn't work, and if it does, we should be releasing 3.21.0 with the fix this week.

**Comment by jkrems** [comment]
It looks like it works! I wasn't able to verify it within the monorepo/the depot but at least the following index.js runs in both node v18.2.0 and v18.8.0:

```js
const tf = require('@tensorflow/tfjs-core');
const {setWasmPath} = require('@tensorflow/tfjs-backend-wasm/dist/tf-backend-wasm.min.js');

const wasmPath = require.resolve('@tensorflow/tfjs-backend-wasm/wasm-out/tfjs-backend-wasm.wasm');
setWasmPath(wasmPath);
(async () => {
  await tf.setBackend('wasm');
  console.log('ok');
})();
```

**Comment by mattsoulanille** [comment]
I think I found out why this is working now and not in tfjs-v3.20.0. `git bisect` tells me node 18 was fixed by cc389ed5a98c6e592a096947bacb864fdf7b323c, which upgraded emsdk to 3.1.20. This makes sense since the error appeared in an emscripten-generated file.

**Comment by mattsoulanille** [comment]
3.21.0, which included the fix for node 18, has been synced to google3, so I'm closing this issue.
